docs: add technical guide for x402 payment integration on Rootstock

swarnasn29 opened the pull request:
## Title

Add technical guide for x402 payment integration on Rootstock

## Description

* **Purpose**: Provides a tutorial for implementing the x402 (Payment Required) standard on Rootstock.
* **Feature**: New guide for building Bitcoin-secured paywalls using tRBTC.
* **Changes**: Added `docs/04-resources/04-tutorials/integrate-x402.md` with full implementation logic using `web3` and Redis.

## Screenshots/GIFs

*N/A (Documentation update)*

## Testing

* Verified with `yarn build` to ensure no broken links or MDX errors.
* Code logic manually tested against Rootstock Testnet (Chain ID 31) for transaction verification.

## Checklist

* [x] I have read and understood the contributing guidelines.
* [x] I have followed the style guide and formatting guidelines.
* [x] I have added appropriate comments to explain the changes.
* [x] I have tested my changes thoroughly.

## Refs

* [x402 Protocol Spec](https://x402.org)

owans reviewed (approved):
LGTM!

swarnasn29 committed:
docs: add technical guide for x402 payment integration on Rootstock

owans commented on docs/04-resources/04-tutorials/index.md:
```diff
@@ -143,5 +143,13 @@ color="orange"
+<FilterItem
+value="beginner, advanced"
+title="Integrating x402 Payments with Rootstock"
+subtitle="Http 402"
```
```suggestion
subtitle="payments"
```

owans commented on docs/04-resources/04-tutorials/integrate-x402.md:
```diff
@@ -0,0 +1,354 @@
+
+## The Protocol Flow 
+
+Unlike hosted solutions, "Sovereign Mode" means your API acts as its own payment processor. This eliminates middleman fees and reliance on third-party gateways.
```
Is sovereign mode already explained somewhere on this guide, if not, link externally or in page

owans commented on docs/04-resources/04-tutorials/integrate-x402.md:
````diff
@@ -0,0 +1,354 @@
+```
+
+
+## 4. Testing the Integration
````
All the above commands are within server.js right? Please add the full code for easy copy and paste

owans commented on docs/04-resources/04-tutorials/integrate-x402.md:
```diff
@@ -0,0 +1,354 @@
+
+## Resources
+
+* **[Rootstock Developers Portal](https://dev.rootstock.io/)** - The central hub for RSK documentation.
```
```suggestion
* **[Rootstock Developers Portal](https://dev.rootstock.io/)** - The central hub for Rootstock documentation.
```

owans commented on docs/04-resources/04-tutorials/integrate-x402.md:
```diff
@@ -0,0 +1,354 @@
+
+### Step 3.4: The Protected Route
+
+Finally, we apply our middleware to the route.
```
```suggestion
Finally, apply the middleware to the route.
```

owans commented on docs/04-resources/04-tutorials/integrate-x402.md:
```diff
@@ -0,0 +1,354 @@
+
+### Step 3.1: Imports and Initialization
+
+First, we set up our environment. We use `express` for the API, `web3` to talk to the Rootstock blockchain, and `redis` to remember which payments have already been spent.
```
```suggestion
First, we set up our environment. We use `express` for the API, `web3` to communicate to the Rootstock blockchain, and `redis` to remember which payments have already been spent.
```

owans commented on docs/04-resources/04-tutorials/integrate-x402.md:
```diff
@@ -0,0 +1,354 @@
+* **Rootstock Testnet Wallet** funded with `tRBTC`.
+    * [**Rootstock Faucet**](https://faucet.rootstock.io/)
+* **RPC Endpoint:** * Testnet: `https://public-node.testnet.rsk.co`
+    * *Recommendation:* For production, use a dedicated RPC key from providers like pure RPC or QuickNode to avoid rate limits.
```
Link to RPC Tools section: https://dev.rootstock.io/dev-tools/node-rpc/

swarnasn29 committed:
docs: apply review suggestions

swarnasn29 commented:
Thanks @owans ! I've applied your suggestions regarding the RPC links and grammar. I also fixed the linting ('RSK' -> 'Rootstock') and added the complete server.js code block as requested. Ready for re-review!

owans commented on docs/04-resources/04-tutorials/integrate-x402.md:
```diff
@@ -0,0 +1,505 @@
+REDIS_URL=redis://127.0.0.1:6379
+
+# Rootstock Testnet RPC
+RSK_NODE_RPC=[https://public-node.testnet.rsk.co](https://public-node.testnet.rsk.co)
```
We recommend using the Rootstock RPC API Testnet URL: https://rpc.rootstock.io/

owans commented on docs/04-resources/04-tutorials/integrate-x402.md:
```diff
@@ -0,0 +1,505 @@
+tags: [rsk, rootstock, tutorials, x402, payments, agents]
+---
+
+# Integrating x402 Payments with Rootstock
```
Please remove this as the meta description already adds a title

owans commented:
> Thanks @owans ! I've applied your suggestions regarding the RPC links and grammar. I also fixed the linting ('RSK' -> 'Rootstock') and added the complete server.js code block as requested. Ready for re-review!

Hi @swarnasn29 

Thank you for making the updates, I dropped two comments, once applied I will merge. Thank you

swarnasn29 committed:
fix: remove redundant H1 and update RPC to use API key format

swarnasn29 commented:
Thanks @owans! I've removed the redundant H1 title and updated the environment variable to use the new Rootstock RPC API endpoint format (https://rpc.testnet.rootstock.io/<API_KEY>).

owans commented on docs/04-resources/04-tutorials/integrate-x402.md:
```diff
@@ -0,0 +1,503 @@
+tags: [rsk, rootstock, tutorials, x402, payments, agents]
+---
+
+The **x402 protocol** (deriving from HTTP Status `402 Payment Required`) is emerging as the standard for **Agentic Commerce**. It allows AI agents, automated scripts, and browsers to autonomously negotiate and pay for resources, such as premium APIs, gated content, or computational tasks, without human intervention.
```
@swarnasn29 Minor feedback:

To improve readability, please reserve backticks ( ) strictly for technical identifiers like function names, variables, file paths, or CLI commands. Overusing inline code for general emphasis can make the page feel cluttered. For other highlights or important terms, please use bold or italics instead.

owans commented on docs/04-resources/04-tutorials/integrate-x402.md:
```diff
@@ -0,0 +1,503 @@
+
+Unlike hosted solutions, "Sovereign Mode" means your API acts as its own payment processor. This eliminates middleman fees and reliance on third-party gateways.
+
+1.  **Challenge (Handshake):** The client requests a resource (e.g., `/api/premium`). The server detects a missing payment header and responds with `402 Payment Required`. Crucially, it returns a `WWW-Authenticate` header containing the **Price**, **Asset (tRBTC)**, and **Target Address**.
```
```suggestion
1.  **Challenge (Handshake):** The client requests a resource (e.g., **/api/premium**). The server detects a missing payment header and responds with _402 Payment Required_. Crucially, it returns a _WWW-Authenticate_ header containing the **Price**, **Asset (tRBTC)**, and **Target Address**.
```

owans replied:
Any update on this @swarnasn29 ?

swarnasn29 committed:
chore: updated markdown formatting to bold/italics per review

swarnasn29 commented:
Thanks @owans! I've updated the formatting to reduce backticks and use bold/italics as requested.

owans commented:
> Thanks @owans! I've updated the formatting to reduce backticks and use bold/italics as requested.

Noted. Thank you

owans reviewed (approved):
LGTM!

Congratulations @swarnasn29,

Your Pull Request Has Been Approved 🎉

I'm happy to let you know that your pull request to add the x402 payment integration on Rootstock guide has been reviewed and successfully merged into the Rootstock Developer Portal 🎉.

Thank you for your valuable contribution to the Rootstock ecosystem and for helping enhance our developer documentation.
